sw/host/tests/crypto/ecdsa_kat/src/main.rs

```diff
@@ -63,6 +72,11 @@ const ECDSA_CMD_MAX_COORDINATE_BYTES_P256: usize = 32;
 const ECDSA_CMD_MAX_SIGNATURE_SCALAR_BYTES_P384: usize = 48;
 const ECDSA_CMD_MAX_COORDINATE_BYTES_P384: usize = 48;
 
+// These values were generated randomly for testing purposes.
+// Each value must be less than the modulus for its respective curve.
+const RANDOM_MASK_P256: &str = "9665cc1c4e9e15354e3fc319ca7de255f2122dc3c16da15e6a83ce1d7a912df0";
```

Comment:
Is there a reason we're doing the masking on the host side? I'm not sure of the performance differences, but if the current bottleneck in these tests is the UART, could we benefit by not sending twice as many bytes for the private key?

Reply:
Unlike in AES, masking for asymmetric crypto is a rather complex operation. The two shards of the private key need to add to the actual key, mod the size of the curve. Here, we use a known-good Rust crypto library to perform the modular subtraction necessary to extract the two shards. If we did this on the device-side, we would have the implement the modular arithmetic entirely from scratch.